src/TestTraits/CliTestTrait.php

```diff
@@ -75,6 +75,42 @@ public function getErrorOutputRaw()
         return $this->process ? $this->process->getErrorOutput() : '';
     }
 
+    /**
+     * Run a command in a child process, and return the process.
+     *
+     * @param string $command
+     *   The actual command line to run.
+     * @param sting cd
+     *   The directory to run the command in.
+     * @param array $env
+     *  Extra environment variables.
+     * @param string $input
+     *   A string representing the STDIN that is piped to the command.
+     */
+    public function startExecute($command, $cd = null, $env = null, $input = null)
```

Comment:
Is there no way to use $this->execute()? Would save a lot of duplicated code.

Reply:
there is a lot of duplicated code, and i'm open to ways to reduce it.

the main new capability needed is 'run child process in the background, without blocking on it's completion, with access to output for tests'.

my first thought was that should be a separate method, rather than another flag and return type from execute(). so i was planning to reduce duplicate code by factoring out the shared bits in execute() and startExecute(). i also don't love the name startExecute(), happy to get better suggestions.

make sense?

Reply:
How about we sadd a $timeout param and stop hardocding `0` as timeout [in this line](https://github.com/drush-ops/drush/blob/e343e01486a740eace4b47a58a849cada8fb6a6b/src/TestTraits/CliTestTrait.php#L102). And we add the same timeout param to drush() method which thencalls execute(). Then the test can just call something like `$this->drush(watchdog:show --tail, $timeout)`

Reply:
I think that's a useful but slightly different improvement.

Too feel confident that this works, I'd like to create multiple watchdog entries while the process running `watchdog:show --tail` is running. That requires interactivity in the test process, so we can't block on the `$this->drush()` call.

Reply:
updated the PR to factor out much of the duplicated code.